substrate/bin/utils/chain-spec-builder/src/lib.rs

```diff
@@ -21,31 +21,76 @@
 //! A chain-spec is short for `chain-configuration`. See the [`sc-chain-spec`] for more information.
 //!
 //! Note that this binary is analogous to the `build-spec` subcommand, contained in typical
-//! substrate-based nodes. This particular binary is capable of building a more sophisticated chain
-//! specification that can be used with the substrate-node, ie. [`node-cli`].
+//! substrate-based nodes. This particular binary is capable of interacting with
+//! [`sp-genesis-builder`] implementation of any provided runtime allowing to build chain-spec JSON
+//! files.
 //!
-//! See [`ChainSpecBuilder`] for a list of available commands.
+//! See [`ChainSpecBuilderCmd`] for a list of available commands.
+//!
+//! ## Typical use-cases.
+//! - **Get default config from runtime.**
+//!
+//!		Query the default genesis config from the provided `runtime.wasm` and use it in the chain
+//! spec. Tool can also store runtime's default genesis config in given file (`-d`):
+//!		```text
+//! 	chain-spec-builder create -r runtime.wasm default -d /dev/stdout
+//! 	```
+//!
+//! 	_Note:_ `GenesisBuilder::create_default_config` runtime function is called.
+//!
+//!
+//! - **Generate raw storage chain spec using genesis config patch.**
+//!
+//!		Patch the runtime's default genesis config with provided `patch.json` and generate raw
+//! storage (`-s`) version of chain spec:
+//!		```text
+//! 	chain-spec-builder create -s -r runtime.wasm patch -p patch.json
+//! 	```
+//!
+//! 	_Note:_ `GenesisBuilder::build_config` runtime function is called.
+//!
+//! - **Generate raw storage chain spec using full genesis config.**
+//!
+//!		Build the chain spec using provided full genesis config json file. No defaults will be used:
+//! 	```text
+//! 	chain-spec-builder create -s -r runtime.wasm full -c full-genesis-config.json
+//! 	```
+//!
+//! 	_Note_: `GenesisBuilder::build_config` runtime function is called.
+//!
+//! - **Generate human readable chain spec using provided genesis config patch.**
+//!
+//!		```text
+//! 	chain-spec-builder create -r runtime.wasm patch -p patch.json
```

Comment:
```suggestion
//!	```text
//! 	chain-spec-builder create -r runtime.wasm patch -p patch.json
```